addons/account/models/account_payment.py

```diff
@@ -18,7 +19,8 @@ class AccountPayment(models.Model):
         string='Journal Entry', required=True, readonly=True, ondelete='cascade',
         index='btree_not_null',
         check_company=True)
-
+    journal_id = fields.Many2one('account.journal', compute='_compute_journal_id', inverse='_inverse_company_journal', store=True, readonly=False, precompute=True)
+    company_id = fields.Many2one('res.company', compute='_compute_company_id', inverse='_inverse_company_journal', store=True, readonly=False, precompute=True)
```

Comment:
Might be good to explicitly mark them as `index=False` and add a comment they are covered by compound indexes in the init. To prevent future addition of an index on `journal_id` which would be useless.

Note that a query with a criteria *only* on the `company_id` will never hit the compound index, as the first key is the `journal_id`, something to take into consideration.